src/psyclone/psyGen.py

```diff
@@ -5984,6 +6001,18 @@ def __init__(self, statements, parent=None):
         else:
             self.ast = None
             self.ast_end = None
+        # Store any Code Block properties.
+        self._structure = structure
```

Comment:
Do we need type checking to verify the argument `structure` is an element or the Enum `Structure`?

Reply:
I thought about this then decided that it is not something that a user would specify, it is something that is created by a front end so thought it would be OK to assume the inputs are correct. However, I can obviously add tests and checks if you think it makes sense.

Reply:
OK, I am happy to assume the inputs are correct in this case.